src/Stratis.Bitcoin.Features.Miner.Tests/PosBlockAssemblerTest.cs

```diff
@@ -0,0 +1,575 @@
+                this.network.Consensus.BIP9Deployments[0] = null;
+            }
+        }
+
+        private static ConcurrentChain GenerateChainWithHeight(int blockAmount, Network network, Key key)
+        {
+            var chain = new ConcurrentChain(network);
+            var nonce = RandomUtils.GetUInt32();
+            var prevBlockHash = chain.Genesis.HashBlock;
+            for (var i = 0; i < blockAmount; i++)
+            {
+                var block = new Block();
+                Transaction coinStake = CreateCoinStakeTransaction(network, key, chain.Height + 1, new uint256((ulong)12312312 + (ulong)i));
+
+                block.AddTransaction(coinStake);
+                block.UpdateMerkleRoot();
+                block.Header.BlockTime = new DateTimeOffset(new DateTime(2017, 1, 1, 0, 0, 0, DateTimeKind.Utc).AddDays(i));
+                block.Header.HashPrevBlock = prevBlockHash;
+                block.Header.Nonce = nonce;
+
+                chain.SetTip(block.Header);
+                prevBlockHash = block.GetHash();
+            }
+
+            return chain;
+        }
+
+        private static ConcurrentChain GenerateChainWithHeightAndActivatedBip9(int blockAmount, Network network, Key key, BIP9DeploymentsParameters parameter, Target bits = null)
+        {
+            var chain = new ConcurrentChain(network);
+            var nonce = RandomUtils.GetUInt32();
+            var prevBlockHash = chain.Genesis.HashBlock;
+            for (var i = 0; i < blockAmount; i++)
+            {
+                var block = new Block();
+                Transaction coinbase = CreateCoinStakeTransaction(network, key, chain.Height + 1, new uint256((ulong)12312312 + (ulong)i));
+
+                block.AddTransaction(coinbase);
+                block.UpdateMerkleRoot();
+                block.Header.BlockTime = new DateTimeOffset(new DateTime(2017, 1, 1, 0, 0, 0, DateTimeKind.Utc).AddDays(i));
+                block.Header.HashPrevBlock = prevBlockHash;
+                block.Header.Nonce = nonce;
+
+                if (bits != null)
+                {
+                    block.Header.Bits = bits;
+                }
+
+                if (parameter != null)
+                {
+                    uint version = ThresholdConditionCache.VersionbitsTopBits;
+                    version |= ((uint)1) << parameter.Bit;
+                    block.Header.Version = (int)version;
+                }
+
+                chain.SetTip(block.Header);
+                prevBlockHash = block.GetHash();
+            }
+
+            return chain;
+        }
+
+        private static Transaction CreateCoinStakeTransaction(Network network, Key key, int height, uint256 prevout)
+        {
+            var coinStake = new Transaction();
+            coinStake.AddInput(new TxIn(new OutPoint(prevout, 1)));
+            coinStake.AddOutput(new TxOut(0, new Script()));
+            coinStake.AddOutput(new TxOut(network.GetReward(height), key.ScriptPubKey));
```

Comment:
I am not sure `network.GetReward(height)` is the correct method for stake trx

Reply:
This is for the blocktemplate. In the actual PosMinting it's this.posConsensusValidator.GetProofOfStakeReward(chainTip.Height) + fees. However that is not what is coming out of the blockassembler. I can change it to Money.Zero or just 0 to more closely reflect what the assembler is actually doing. What do you think?

Reply:
hmm but then it wont pass the consensus checks right?

Reply:
As these are unit tests the consensus checks are mocked in this code so it will pass. It however won't pass if we run this with consensus checks enabled.